shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Coin.hs

```
@@ -47,18 +51,16 @@ coinToRational (Coin c) = fromIntegral c
 rationalToCoinViaFloor :: Rational -> Coin
 rationalToCoinViaFloor r = Coin . floor $ r
 
-isValidCoinValue :: Integer -> Bool
-isValidCoinValue c = 0 <= c && c <= (fromIntegral (maxBound :: Word64))
+-- FIXME: handle coin out of bounds. where?
+-- if coin is less than 0 or greater than (maxBound :: Word64), then
+-- fromIntegral constructs the incorrect value
+instance Core.Compactible Coin where
+  newtype CompactForm Coin = CompactCoin Word64
+  toCompact = CompactCoin . fromIntegral . unCoin
```

Maybe we want to throw an error here if the coin isn't valid, since we never want to compact an "intermediate" coin